src/System.Management.Automation/FormatAndOutput/common/Utilities/MshObjectUtil.cs

```diff
@@ -290,10 +291,8 @@ internal static string SmartToString(PSObject so, PSPropertyExpressionFactory ex
                 // take care of the case there is no base object
                 return so.ToString();
             }
-            catch (ExtendedTypeSystemException e)
+            catch (Exception e) when (e is ExtendedTypeSystemException || e is InvalidOperationException)
             {
-                // NOTE: we catch all the exceptions, since we do not know
-                // what the underlying object access would throw
```

Comment:
Maybe update the comment? It would be nice for readers to understand why we catch these two exceptions.

Reply:
will update

Reply:
Since we silently return empty string it is difficult to discover for users why they get empty string instead of object list. I suggest add a trace/log here.

Reply:
added trace